### ISSUE TITLE
Line-breaks in a Delta Format

### ISSUE THREAD

**Issue by npcole** [quilljs/quill#2469]
In common with #1511 and also with #252, I think there is a compelling reason to decide on a way to represent a line-break within a line.  Even within a heading, it may be necessary to specify a line-break that is not in fact a paragraph break, but for many applications, the ability to have line-breaks within table cells, list items, etc. is important.  In other words, the need is more general than the one outlined by #1511 .

The easiest way to represent this would seem to be to pick a character to represent a line-break.  In #252 someone suggested the Unicode \u000b character, which seems appropriate.

As for a compelling use-case: I'm looking at adapting quilljs to work as an editor for TEI -- and in the document-editing world it is absolutely essential to be able to distinguish line-breaks from paragraph blocks.

From a technical point of view, is there any reason not to adopt this proposal?  A line-break character could be treated by delta formats as just another character, couldn't it? There's no reason to treat it as special in any way that I can see.  However, if there is a technical reason this is difficult, I'd be really interested in understanding it.

**Comment by ile** [comment]
I made this, maybe it helps someone: https://gist.github.com/ile/ce2e56b2d96a5e2c66e04b0ac5e4d3f3

**Comment by ile** [comment]
I made this "mod": https://gist.github.com/ile/8a08c23b5ae0f331dee8ca062fdef40d 

It requires my fork of Quill, which has only a little change, but it is unlikely it will be incorporated into the Quill itself. https://github.com/ilkkah/quillty

It "seems" to work but I cannot guarantee 100 % perfect operation.

**Comment by ile** [comment]
I made this and it seems to work to some extent: https://gist.github.com/ile/8a08c23b5ae0f331dee8ca062fdef40d

**Comment by martijndekuijper** [comment]
I still can't believe line-breaks are not supported :(

@ile you say:
> To some extent

What are the limitations?

**Comment by ile** [comment]
> What are the limitations?

It's a bit of a guesswork. And it needs my fork of Quill which has a "dirty hack".

There are no real tests to verify that it works 100 %. And I noticed that it seems to get confused if there are a lot of styles and undo/redo is applied. "A lot of styles" means here that I tested it with a template from Google Docs, edited that and tried undo/redo.

But the problems were quite minor that I could find so at the moment I'm using it. Improvements are welcome of course.
